barretenberg/cpp/src/barretenberg/relations/ecc_vm/ecc_set_relation_impl.hpp

```diff
@@ -391,25 +392,28 @@ void ECCVMSetRelationImpl<FF>::accumulate(ContainerOverSubrelations& accumulator
 {
     using Accumulator = typename std::tuple_element_t<0, ContainerOverSubrelations>;
     using View = typename Accumulator::View;
+    using ShortView = typename std::tuple_element_t<1, ContainerOverSubrelations>::View;
 
     // degree-11
     Accumulator numerator_evaluation = compute_grand_product_numerator<Accumulator>(in, params);
 
-    // degree-17
+    // degree-20
     Accumulator denominator_evaluation = compute_grand_product_denominator<Accumulator>(in, params);
 
     const auto& lagrange_first = View(in.lagrange_first);
     const auto& lagrange_last = View(in.lagrange_last);
+    const auto& lagrange_last_short = ShortView(in.lagrange_last);
 
     const auto& z_perm = View(in.z_perm);
     const auto& z_perm_shift = View(in.z_perm_shift);
+    const auto& z_perm_shift_short = ShortView(in.z_perm_shift);
 
-    // degree-18
+    // degree-21
     std::get<0>(accumulator) +=
         ((z_perm + lagrange_first) * numerator_evaluation - (z_perm_shift + lagrange_last) * denominator_evaluation) *
         scaling_factor;
 
     // Contribution (2)
-    std::get<1>(accumulator) += (lagrange_last * z_perm_shift) * scaling_factor;
+    std::get<1>(accumulator) += lagrange_last_short * z_perm_shift_short * scaling_factor;
```

Comment:
Small optimization: taken into account the fact that the second component of the accumulator is actually of a much smaller degree ( 2 )